api/graphql/resolvers/user.go

```diff
@@ -316,46 +321,61 @@ func (r *mutationResolver) UserRemoveRootAlbum(ctx context.Context, userID int,
 		}
 
 		// Cleanup if no user owns the album anymore
-		var userAlbumCount int
-		if err := tx.Raw("SELECT COUNT(user_id) FROM user_albums WHERE album_id = ?", albumID).Scan(&userAlbumCount).Error; err != nil {
+		deletedAlbumIDs, err = cleanup(tx, albumID, deletedAlbumIDs, childAlbumIDs)
+		if err != nil {
 			return err
 		}
 
-		if userAlbumCount == 0 {
-			deletedAlbumIDs = append(childAlbumIDs, albumID)
-			childAlbumIDs = nil
-
-			// Delete albums from database
-			if err := tx.Delete(&models.Album{}, "id IN (?)", deletedAlbumIDs).Error; err != nil {
-				deletedAlbumIDs = nil
-				return err
-			}
-		}
-
 		return nil
 	})
 
 	if transactionError != nil {
 		return nil, transactionError
 	}
 
+	if err := clearCacheAndReloadFaces(deletedAlbumIDs, db); err != nil {
+		return nil, err
+	}
+
+	return &album, nil
+}
+
+func cleanup(tx *gorm.DB, albumID int, deletedAlbumIDs []int, childAlbumIDs []int) ([]int, error) {
```

Comment:
I can't read this function quickly. What's the difference between `albumID` and `deletedAlbumIDs`?

Reply:
Hmm, maybe I don't understand some aspects of your question, but it looks pretty straightforward:
`albumID int` - passed as a param to the parent function; represents the current album, we're working on; is appended to the array of `deletedAlbumIDs` if `userAlbumCount == 0`
`deletedAlbumIDs []int` - a local var in the parent function, representing an array of deleted albums; passed as a param to this function, where updated and returned back to the parent function; in the scope of this function it can grow, appending the `albumID` if its `userAlbumCount == 0`

BTW, probably line 356 `deletedAlbumIDs = nil` can be safely removed, as it assigns `nil` to the variable, which would be never used. What do you think?

Reply:
`deletedAlbumIDs, err = cleanup(tx, albumID, deletedAlbumIDs, childAlbumIDs)` deletedAlbumIDs is both input and output parameter. It's not a good design.

I guess `deletedAlbumIDs` should be created by this function and returned to the caller.

Reply:
You're right, thanks. The `deletedAlbumIDs` is declared but never updated before calling the `cleanup()` in the caller function, so it doesn't make sense to pass an empty var from the caller to the function.
Updated the code.